server/src/GraphQL/Queries/Inspector.php

```diff
@@ -0,0 +1,186 @@
+            "events" => $eventMapping,
+            "publishable_providers" => $publishableProviders,
+            "laravel_log" => $logFileNames
+        ];
+
+        return json_encode($packageDataStructure);
+    }
+
+    public function logContents($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $fileToLookFor = $args['filename'];
+        $returnArray = [];
+        foreach (glob(storage_path() . '/logs/*.log') as $filename) {
+            if (basename($filename) == $fileToLookFor) {
+                $file = File::get($filename);
+                $parsed = $this->parseLogFile($file);
+                array_push($returnArray, $fileToLookFor);
+                array_push($returnArray, $parsed);
+            }
+        }
+        return json_encode($returnArray);
+    }
+
+    public function filteredLogContents($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $incomingFileNames = $args['filenames'];
+        $keyword = $args['keyword'];
+        $result = [];
+
+        // Parse all files that have been specified in the args if they contain the keyword
+        foreach (glob(storage_path() . '/logs/*.log') as $filename) {
+            if (in_array(basename($filename), $incomingFileNames)) {
+                $file = File::get($filename);
+                if (preg_match("/{$keyword}/i", $file)) {
+                    array_push($result, basename($filename));
+                }
+            }
+        }
+
+        return json_encode($result);
+    }
+
+
+    public function parseLogFile($file)
+    {
+        $message_levels = [
+            'debug' => 'info-circle',
+            'info' => 'info-circle',
+            'notice' => 'info-circle',
+            'warning' => 'exclamation-triangle',
+            'error' => 'exclamation-triangle',
+            'critical' => 'exclamation-triangle',
+            'alert' => 'exclamation-triangle',
+            'emergency' => 'exclamation-triangle',
+            'processed' => 'info-circle',
+            'failed' => 'exclamation-triangle'
+        ];
+
+        $log = [];
+
+        preg_match_all('/\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}([\+-]\d{4})?\].*/', $file, $headings);
+
+        if (!is_array($headings)) {
+            return $log;
+        }
+
+        $log_data = preg_split('/\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}([\+-]\d{4})?\].*/', $file);
+
+        if ($log_data[0] < 1) {
+            array_shift($log_data);
+        }
+
+        foreach ($headings as $h) {
+            for ($i = 0, $j = count($h); $i < $j; $i++) {
+                foreach (array_keys($message_levels) as $level) {
+                    if (strpos(strtolower($h[$i]), '.' . $level) || strpos(strtolower($h[$i]), $level . ':')) {
+                        preg_match('/^\[(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}([\+-]\d{4})?)\](?:.*?(\w+)\.|.*?)' . $level . ': (.*?)( in .*?:[0-9]+)?$/i', $h[$i], $current);
```

Comment:
@AlexeiaW so regex will be a good area to test in general. It may be better to split this out into a function that you can separately test, to ensure the regex matches. 

Additionally you can test the entire method by using a mock log file with various inputs you might expect from third party packages, you can even test unusual characters, exclamation points or other characters etc.

Reply:
Ok, I made a make log file that contains various log messages and use that to test the function that splits up the log file